source/docs/flow_control.md

```diff
@@ -0,0 +1,103 @@
+### Overview
```

Comment:
Thanks this is very useful. nit: it would be easier to read if using `foo` for functions and possibly using some numeric lists/bullet lists in subsections.